### PR TITLE
8330806: test/hotspot/jtreg/compiler/c1/TestLargeMonitorOffset.java fails on ARM32

### PR DESCRIPTION
**Pull request by voitylov**
TestLargeMonitorOffset was introduced by 8310844 with a fix for the AArch64 platform. The same issue needs to be fixed for ARM32. With this change, we add the large slot_offset handling to the ARM32 version of IR_Assembler::osr_entry().

Testing: jtreg hotspot, jtreg jdk tier1-3.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330806](https://bugs.openjdk.org/browse/JDK-8330806): test/hotspot/jtreg/compiler/c1/TestLargeMonitorOffset.java fails on ARM32 (**Bug** - P3)

### Reviewers
 * [Sergey Nazarkin](https://openjdk.org/census#snazarki) (@snazarkin - no project role)
 * [Dmitry Samersoff](https://openjdk.org/census#dsamersoff) (@dsamersoff - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18891/head:pull/18891` \
`$ git checkout pull/18891`

Update a local copy of the PR: \
`$ git checkout pull/18891` \
`$ git pull https://git.openjdk.org/jdk.git pull/18891/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18891`

View PR using the GUI difftool: \
`$ git pr show -t 18891`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18891.diff">https://git.openjdk.org/jdk/pull/18891.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18891#issuecomment-2069654619)

### CODE REVIEW

File: src/hotspot/cpu/arm/c1_LIRAssembler_arm.cpp [hunk 1]
```diff
@@ -152,8 +152,14 @@ void LIR_Assembler::osr_entry() {
   int monitor_offset = (method()->max_locals() + 2 * (number_of_locks - 1)) * BytesPerWord;
   for (int i = 0; i < number_of_locks; i++) {
     int slot_offset = monitor_offset - (i * 2 * BytesPerWord);
-    __ ldr(R1, Address(OSR_buf, slot_offset + 0*BytesPerWord));
-    __ ldr(R2, Address(OSR_buf, slot_offset + 1*BytesPerWord));
+    if (slot_offset >= 4096 - BytesPerWord) {
+      __ add_slow(R2, OSR_buf, slot_offset);
```
Comment (snazarkin) [inline src/hotspot/cpu/arm/c1_LIRAssembler_arm.cpp:156]: Can't we check this once before the loop? Or doesn't it just make sense?

Comment (bulasevich) [reply src/hotspot/cpu/arm/c1_LIRAssembler_arm.cpp:156]: Hi Sergey. Thanks for looking at this.
This is not performance critical code, and the typical number_of_locks value is 0, so IF inside the FOR loop makes sense here.

**Comment by snazarkin** [comment]
I've checked the patch (one may need to use a [workaround](https://bugs.openjdk.org/browse/JDK-8316395) ).  The JDK crashes without the patch, and passes with the patch.

**Review by snazarkin — APPROVED** [review]

**Comment by dsamersoff** [comment]
Looks good for me.

**Review by dsamersoff — APPROVED** [review]
Looks good to me

**Comment by bulasevich** [comment]
Looks good

**Pull request closed without merging**